File: python/58.length-of-last-word.py
#
# @lc app=leetcode id=58 lang=python3
#
# [58] Length of Last Word
#

# @lc code=start
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        return len(s.rstrip().rsplit(' ',maxsplit=1)[-1])
# @lc code=end

File: rust/58.length-of-last-word.rs
/*
 * @lc app=leetcode id=58 lang=rust
 *
 * [58] Length of Last Word
 */

// @lc code=start
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        s.trim()
            .split(' ')
            .last()
            .map(|x| x.len() as i32)
            .unwrap()
    }
}
// @lc code=end
